state-chain/traits/src/lib.rs

```diff
@@ -192,7 +192,12 @@ pub trait VaultActivator<C: ChainCrypto> {
 
 	/// Activate key/s on particular chain/s. For example, setting the new key
 	/// on the contract for a smart contract chain.
-	fn activate(new_key: C::AggKey, maybe_old_key: Option<C::AggKey>);
+	fn activate(
+		new_key: C::AggKey,
+		maybe_old_key: Option<C::AggKey>,
+	) -> Option<ThresholdSignatureRequestId>;
+
+	fn activate_key();
```

Comment:
Doc comment would be nice.

Reply:
Maybe rename `activate` -> `start_key_activation`.

Reply:
I added something, not sure how specific I have to be.
Is it ok like this?